corehq/form_processor/migrations/sql_templates/migrate_balance_2_create_trigger.sql

```diff
@@ -0,0 +1,9 @@
+CREATE FUNCTION copy_balance() RETURNS trigger AS $copy_balance$
+    BEGIN
+        NEW.new_updated_balance = NEW.updated_balance;
+        RETURN NEW;
+    END;
+$copy_balance$ LANGUAGE plpgsql;
+
+CREATE TRIGGER copy_balance_trigger BEFORE INSERT OR UPDATE ON form_processor_ledgertransaction
+    FOR EACH ROW EXECUTE PROCEDURE copy_balance();
```

Comment:
Curious why the trigger is necessary? Does the backfill not already take care of this? (assuming you do not use the `WHERE` condition I proposed)

Assuming the trigger is necessary for some reason I'm not aware of, could this file be merged with `migrate_balance_1_add_column.sql`?

Reply:
Update: oh, I think I get it. It's so new rows added after the backfill transaction starts are updated correctly and do not cause the `NOT NULL` constraint to fail. Cool.